Example/Example/ViewControllers/DemoViewController.swift

```diff
@@ -85,6 +103,23 @@ extension DemoViewController {
         }
     }
 
+    fileprivate func requestEngagement(selectedOperator: Operator) {
+        configureInteractor(chatType: .sync)
+
+        Salemove.sharedInstance.requestEngagement(with: selectedOperator) { [unowned self] engagementRequest, error in
+            if let reason = error?.reason {
+                self.showError(message: reason)
+            } else if let request = engagementRequest {
+                guard let controller = self.statusViewController else {
```

Comment:
What is this for?

Reply:
controller can be nil when not configured

Reply:
You configure it just few lines above

Reply:
Doesnt matter we are inside a callback and this callback will be called after some period of time. As we specify that the status view controller is optional then we need to first check that it's present. I would avoid force unwrap here.

Reply:
Well, what happens if it's not there? The engagement request will just continue?
The engagement request should be at least canceled.